src/main/java/bio/terra/service/filedata/google/firestore/FireStoreBatchQueryIterator.java

```diff
@@ -59,26 +63,20 @@ public FireStoreBatchQueryIterator(Query baseQuery, int batchSize) {
     logger.info("Retrieving batch {} with batch size of {}", count, batchSize);
     count++;
 
-    for (int i = 0; i < RETRIES; i++) {
-      ApiFuture<QuerySnapshot> querySnapshot = query.get();
+    currentList =
+        fireStoreUtils.runTransactionWithRetry(
+            firestore,
+            xn -> {
+              ApiFuture<QuerySnapshot> querySnapshot = xn.get(query);
+              return querySnapshot.get().getDocuments();
+            },
```

Comment:
This is just a style thing but I think it's clear enough to skip the intermediate variable. That lets you get rid of the `{` `}` too.
```suggestion
            xn -> xn.get(query).get().getDocuments(),
```